content/1_General/Generic_Code.mdx

```diff
@@ -327,65 +382,95 @@ of classes and structs, integral to generic code.
 then be forced to create a secondary point struct `pt1`:
 
 ```cpp
-struct pt { int x, y, z; } P1{1, 2, 3}, P2{3, 4, 5};
-struct pt1 { double x, y, z; } P3{1.1, 2.2, 3.3};
+struct Point3D {
+	int x;
+	int y;
+	int z;
+} p1{1, 2, 3}, p2{3, 4, 5};
+
+struct Point3DDouble {
+	double x;
+	double y;
+	double z;
+} p3{1.1, 2.2, 3.3};
 ```
 
 This might not seem too bad immediately, but imagine having to create structs
 like this over and over again just to accommodate various type changes. We need
 something better.
 
-Lo and behold, the **template** to the rescue. We can use the format
+Lo and behold, we have the **template** to come to our rescue. We can use the format
 `template<...>` to specify the specific templating conditions and then simply
 define the struct normally. In particular, if we use a class `T`, we achieve:
 
 ```cpp
-template<class T> struct pt { T x, y, z; };
-pt<int> P1{1, 2, 3}, P2{3, 4, 5};
-pt<double> P3{1.1, 2.2, 3.3};
-pt<long long> P4{9223372036854775807, 9223372036854775807, 9223372036854775807};
+template<class T>
+struct Point3D {
+	T x;
+	T y;
+	T z;
+};
+
+Point3D<int> p1{1, 2, 3};
+Point3D<int> p2{3, 4, 5};
+Point3D<double> p3{1.1, 2.2, 3.3};
+Point3D<long long> p4{
+	9223372036854775807, 9223372036854775807, 9223372036854775807
+};
 ```
 
 It would be narrow-minded to think that templates are in any way limited to
 classes and structs. They can be used with functions and much more. For example,
-
-```cpp
-template<class T> bool ckmin(T& a, const T& b) {
-	return b < a ? a = b, 1 : 0; }
+take a look at this function `ckmin`:
+
+```cpp
+/**
+ * If b is less than a, this changes the value of a to that of b
+ * and returns true. If not, the function simply returns false.
+ */
+template<class T>
+bool ckmin(T& a, const T& b) {
+	if (b < a) {
+		a = b;
+		return true;
+	}
+	return false;
+}
 ```
 
-will set $a$ equal to $\min(a,b)$, and return `true` if the value of $a$
-changes.
-
 One interesting use case is that of the size of various containers. The size
-member function of a container usually returns a type incompatible with integer,
-but we can easily write a templated function to fix this, handling all class
-types `T` of container at once:
+member function of a container usually returns a type incompatible with `int`,
+but we can easily write a templated function to fix this, handling all types
+of containers at once:
 
 ```cpp
-template<class T> inline int sz(T container) {
-	return (int)container.size();
+template<class T>
+int sz(T container) {
+	return (int) container.size();
 }
 ```
 
-You may apply this as `sz<vector<int>>(v)` on some `vector<int> v`, but since
-C++11, functions \(but not classes or structs until C++14 and C++17\) can
-actually infer template arguments, meaning that we can simply use `sz(v)`!
+You can call this through typing `sz<vector<int>>(v)` where `v` is a
+`vector<int>`, but since C++ 11, functions (but not classes or structs
```

Comment:
C++11 shouldn't have a space

Reply:
uh can i have a source for this? willing to change tho

Reply:
look up C++11

Reply:
alr cool